django/db/models/sql/subqueries.py

```diff
@@ -169,3 +179,26 @@ class AggregateQuery(Query):
     def __init__(self, model, inner_query):
         self.inner_query = inner_query
         super().__init__(model)
+
+
+class CheckQuery(Query):
+    compiler = "SQLCheckCompiler"
+
+    def __init__(self):
+        super().__init__(model=None)
+
+    def do_check(self, q, against, using):
+        from django.db.models import Value
+
+        for name, value in against.items():
+            if not hasattr(value, "resolve_expression"):
+                value = Value(value)
+            self.add_annotation(value, name, select=False)
+        # This will raise a FieldError if a field is missing in "against".
+        self.add_q(q)
+        compiler = self.get_compiler(using=using)
```

Comment:
No sure about which parts should remain in `Q.checks` instead. Current separation seems relatively good but I'd be curious about input from others.